src/binary/ContractionBackend.cpp

```diff
@@ -0,0 +1,419 @@
+                                                  std::vector< exec_t >  const & i_exec_type,
+                                                  std::vector< int64_t > const & i_dim_sizes,
+                                                  std::vector< int64_t > const & i_strides_left,
+                                                  std::vector< int64_t > const & i_strides_right,
+                                                  std::vector< int64_t > const & i_strides_out_aux,
+                                                  std::vector< int64_t > const & i_strides_out,
+                                                  data_t                         i_dtype_left,
+                                                  data_t                         i_dtype_right,
+                                                  data_t                         i_dtype_comp,
+                                                  data_t                         i_dtype_out,
+                                                  kernel_t                       i_ktype_first_touch,
+                                                  kernel_t                       i_ktype_main,
+                                                  kernel_t                       i_ktype_last_touch,
+                                                  int64_t                        i_num_threads ){
+
+  //copy to local variables
+  m_dim_type        = i_dim_type;
+  m_exec_type       = i_exec_type;
+  m_dim_sizes       = i_dim_sizes;
+  m_strides_left    = i_strides_left;
+  m_strides_right   = i_strides_right;
+  m_strides_out     = i_strides_out;
+  m_strides_out_aux = i_strides_out_aux;
+  
+  m_dtype_left  = i_dtype_left;
+  m_dtype_right = i_dtype_right;
+  m_dtype_out   = i_dtype_out;
+  m_dtype_comp  = i_dtype_comp;
+
+  m_ktype_first_touch = i_ktype_first_touch;
+  m_ktype_main        = i_ktype_main;
+  m_ktype_last_touch  = i_ktype_last_touch;
+
+  m_num_threads = i_num_threads;
+}
+
+void einsum_ir::binary::ContractionBackend::init( std::vector< iter_property > const & i_loops,
+                                                  data_t                               i_dtype_left,
+                                                  data_t                               i_dtype_right,
+                                                  data_t                               i_dtype_comp,
+                                                  data_t                               i_dtype_out,
+                                                  kernel_t                             i_ktype_first_touch,
+                                                  kernel_t                             i_ktype_main,
+                                                  kernel_t                             i_ktype_last_touch,
+                                                  int64_t                              i_num_threads ){
+
+  int64_t l_num_loops = i_loops.size();
```

Comment:
Fix: ``l_num_loops`` is misleading as not everything might be realized as loops.